### PR TITLE
social media auth

### PR DESCRIPTION
**Pull request by Oluwaseyi000**
### Description
social media authentication enables a user to login and signs up using social media 

### Type of change

- [ ] New Feature (a non-breaking change that adds functionality)

### How Has This Been Tested?

- [ ] Integration

### PT
[#164796851](https://www.pivotaltracker.com/story/show/164796851)

### Thought Process

1. User click on social media button on either  login or signup page, 

Screenshot 2019-05-14 at 10 41 15 AM

2. this redirect them to their social media proprietor pages to enable social media authentication
Screenshot 2019-05-14 at 10 39 28 AM

3. User are redirected to the home page on successfull login or signup
Screenshot 2019-05-14 at 10 43 19 AM

4. The user gets redirected back to login page if the authentication fails
Screenshot 2019-05-14 at 10 52 59 AM

### CODE REVIEW

File: src/__tests__/SocialAuth.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  // const onClick = jest.fn();
+  // const { getByTestid } 
+  const socialButton = render(<SocialButton />);
+  expect(socialButton).toBeTruthy();
+  expect(socialButton).toMatchSnapshot();
+});
+
+describe('action test', () => {
+  it('dispatch acition to social', () => {
+    const action = getSocialAuth(testToken);
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/__tests__/SocialAuth.test.js]: outdated

File: src/__tests__/Home.test.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+import configureMockStore from 'redux-mock-store';
 import Home from '../pages/Index';
+import { renderWithRedux } from '../lib/testUtil';
 
 afterEach(cleanup);
 
+
 test('<Home /> component', () => {
-  const homePage = render(<Home />);
+  const homePage = renderWithRedux(<Home />);
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/__tests__/Home.test.js:11]: resolved

File: src/__tests__/Home.test.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
 import React from 'react';
-import { render, cleanup } from 'react-testing-library';
+import { cleanup } from 'react-testing-library';
+import configureMockStore from 'redux-mock-store';
 import Home from '../pages/Index';
+import { renderWithRedux } from '../lib/testUtil';
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/__tests__/Home.test.js:5]: resolved

File: src/__tests__/Home.test.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
 import React from 'react';
-import { render, cleanup } from 'react-testing-library';
+import { cleanup } from 'react-testing-library';
+import configureMockStore from 'redux-mock-store';
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/__tests__/Home.test.js:3]: resolved

File: src/state/auth/action.js [hunk 5]
```diff
@@ -29,3 +32,27 @@ export const register = newUser => async dispatch => {
...
+
+export const signInError = signinError => ({
+  type: SIGN_IN_ERROR,
+  payload: signinError
+});
+
+export const socialAuth = (token, redirect) => dispatch => {
+  try {
+    setLocalStorage(token, 'loggedinUser');
+    const decodedToken = decodeToken(token);
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/state/auth/action.js:74]: resolved

File: src/state/auth/action.js [hunk 6]
```diff
@@ -29,3 +32,27 @@ export const register = newUser => async dispatch => {
...
+    payload: decodedToken
+  }
+);
+
+export const signInError = signinError => ({
+  type: SIGN_IN_ERROR,
+  payload: signinError
+});
+
+export const socialAuth = (token, redirect) => dispatch => {
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/state/auth/action.js:71]: resolved

File: src/state/auth/action.js [hunk 7]
```diff
@@ -29,3 +32,27 @@ export const register = newUser => async dispatch => {
...
 };
+
+export const getSocialAuth = decodedToken => (
+  {
+    type: SOCIAL_AUTH,
+    payload: decodedToken
+  }
+);
+
+export const signInError = signinError => ({
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/state/auth/action.js:48]: resolved

File: src/state/auth/reducer.js [hunk 8]
```diff
@@ -1,5 +1,7 @@
 import { initialState } from './state';
-import { REGISTER_REQUEST, REGISTER_SUCCESS, REGISTER_FAILURE } from './actionTypes';
+import {
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/state/auth/reducer.js:2]: resolved

File: src/lib/social.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import jwtDecode from 'jwt-decode';
+
+export const redirectSocialAuth = type => {
+  window.location.href = `https://author-haven-stage.herokuapp.com/api/v1/auth/${type}`;
+};
+
+export const decodeToken = token => jwtDecode(token);
+
+export const setLocalStorage = (item, value) => localStorage.setItem(item, value);
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/lib/social.js]: outdated

File: src/__tests__/SocialAuth.test.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    expect(action.type).toEqual('SOCIAL_AUTH');
+    expect(action.payload).isObject;
+  });
+
+  it('test social dispatcher', async () => {
+    const action = await socialAuth(testToken)(socialAuth);
+    expect(action);
+  });
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/__tests__/SocialAuth.test.js]: Outdated

File: src/pages/App.jsx [hunk 11]
```diff
@@ -17,6 +18,8 @@ const App = () => (
       <Route path="/signup" component={Register}></Route>
       <Route path="/profile" exact component={Profile}></Route>
       <Route path="/forgot-password" component={ForgotPassword}></Route>
+      {/* <Route path="/dashboard" exact component={Index}></Route> */}
```
Comment (toluola) [inline src/pages/App.jsx]: Please why the comment?

**Comment by micah-akpan** [comment]
@Oluwaseyi000 please squash your commits

File: src/pages/App.jsx (hunk 11 above)
Comment (Oluwaseyi000) [reply src/pages/App.jsx]: it is an oversight, I have removed it now thanks for spotting it

**Comment by Oluwaseyi000** [comment]
> @Oluwaseyi000 please squash your commits

commits are automatically squash when TTL want to merge, hence I feel no need to manually do it

File: src/components/HandleSocialAuth.jsx [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { socialAuth } from '../state/auth/action';
+
+
+export const handleSocialAuth = props => {
```
Comment (micah-akpan) [inline src/components/HandleSocialAuth.jsx]: Can you make `handleSocialAuth' PascalCase, since it's a component?

Comment (Oluwaseyi000) [reply src/components/HandleSocialAuth.jsx]: thanks for your feedback, done. Thanks

**Review by micah-akpan — DISMISSED** [review]

**Review by Temmyogunbo — DISMISSED** [review]

**Review by Temmyogunbo — APPROVED** [review]

**Pull request merged**